Rakefile

```diff
@@ -27,15 +73,26 @@ namespace :codegen do
           else
             curl -OL https://github.com/google/protobuf/releases/download/v#{protoc_legacy_version}/protoc-#{protoc_legacy_version}-linux-x86_64.zip
           fi
-          unzip protoc-#{protoc_legacy_version}*.zip
+          mkdir -p #{toolchain_bin_dir}
+          unzip -o protoc-#{protoc_legacy_version}*.zip
           mv bin/protoc #{protoc_legacy_binary}
+
+          mkdir -p #{toolchain_legacy_include_dir}
+          rm -rf #{toolchain_legacy_include_dir}/**
+          mv include/* #{toolchain_legacy_include_dir}/
```

Comment:
Nit: decent amount of these lines could be Ruby primitives.

Reply:
This is true, however I find mixing these big sh<<EOD blocks with ruby primitives clumsy.  I can open a follow up to convert more of this to ruby primitives, but chose to leave the big shell blocks as shell to keep the PR smaller.